src/frontend/src/instance.rs

```diff
@@ -396,6 +396,9 @@ impl Instance {
                 // add doc links in err msg later
                 ensure!(!(hit_legacy && hit_prom), OtlpMetricModeIncompatibleSnafu);
 
+                // drop hit_cache to release references before inserting
```

Comment:
Do we need to wrap the `fast cache check` into a method and add a unit test for it?